internal/lib/lib.go

```diff
@@ -1168,7 +1168,7 @@ func ValidateIngressForClass(key string, ingress *networkingv1beta1.Ingress) boo
 	return true
 }
 
-func filterIngressOnClassAnnotation(key string, ingress *networkingv1beta1.Ingress) bool {
+func filterIngressOnClassAnnotation(key string, ingress *networkingv1.Ingress) bool {
 	// If Avi is not the default ingress, then filter on ingress class.
 	if !GetDefaultIngController() {
 		annotations := ingress.GetAnnotations()
```

Comment:
@chauhanshubham : It seems annotation ""kubernetes.io/ingress.class"" has been deprecated and replaced by "spec.IngressClassName" field https://kubernetes.io/blog/2021/07/14/upcoming-changes-in-kubernetes-1-22/#what-to-do. I think we need to change all occurrences for this.

Reply:
done

Reply:
yes! done. thanks.
removed annotation based checks from AKO in the latest commit